File: python/data_structures_fundamentals_edX/problem_1_1_check_brackets_in_the_code/check_brackets_in_the_code.py
code_to_check = "foo()[](bar[i);"


def main():
    print(check_brackets(code_to_check))


def check_brackets(string_of_code):
    stack = []
    count = 0
    for char in code_to_check:
        count += 1
        if char in '([{':
            stack.append(char)
        else:
            if char not in ')]}':
                continue
            if len(stack) == 0:
                return count
            top = stack.pop()
            if (top == '[' and char != ']') or (top == '(' and char != ')') or (top == '{' and char != '}'):
                return count
    return "Success"


if __name__ == '__main__': main()

File: java/data_structures_fundamentals_edX/problem_1_1_check_brackets_in_the_code/src/CheckBracketsInTheCode.java
public class CheckBracketsInTheCode {
    public static void main(String[] args) {
        String code_to_check = "foo()[](bar[i);";
        System.out.println(check_brackets(code_to_check));
    }

    public static String check_brackets(String string_of_code) {
        char stack[];
        int missing_bracket_place_no = 0;
        for  (int i : stack) {
            missing_bracket_place_no++;
            if (i in '([{') {
                stack.append(char);
            } else {
                if (i not in ')]}') {
                    continue;
                };
                if len(stack) == 0 {
                    return missing_bracket_place_no;
                };
                top = stack.pop();
                if (top == '[' && char != ']') || (top == '(' && char != ')') || (top == '{' && char != '}') {
                    return missing_bracket_place_no;
                }
            }
        }
        return "Success";
    }
}
